File: BetterPlacemaking.CLIENT/src/app/models/jetson-dtos/HealthReport.ts
export interface HealthReport {
    Timestamp: number;
    Services?: Record<string, ServiceStatus> | null;
	Cameras?: Record<string, CameraInfo> | null;
    System?: SystemInfo | null;
    IntrinsicsCalibration?: Record<string, IntrinsicsCalibrationState> | null;

    Lidars?: Record<string, LidarInfo> | null;
    PiCompanion?: PiCompanionInfo | null;
}

export interface ServiceStatus {
	Active?: string | null;
	Sub?: string | null;
}

export interface CameraInfo {
	Mac?: string | null;
	Ip?: string | null;
	Resolution?: number[] | null;
	Enabled: boolean;
}

export interface GpuInfo {
	UtilizationPct: number;
	FrequencyMhz: number;
	TemperatureC: number;
}

export interface MemoryInfo {
	UsedMb: number;
	TotalMb: number;
}

export interface SystemInfo {
	Gpu?: GpuInfo | null;
	Memory?: MemoryInfo | null;
	Disk?: DiskPartitionInfo[] | null;
	CpuTemperatureC?: number | null;
}

export interface DiskPartitionInfo {
	Path: string;
	TotalMb: number;
	UsedMb: number;
	FreeMb: number;
	UsePct: number;
	Status: string;
	DeletedFiles: number;
}

export interface IntrinsicsCalibrationState {
	Status: string;
	SightingsCollected: number;
	CoverageGrid: number[];
	SuggestedRegion?: string | null;
	SuggestedTilt?: string | null;
	CurrentRmse: number;
}

export interface LidarInfo {
    SensorId?: string | null;
    Connected: boolean;
    DevicePath?: string | null;
    UsbHint?: string | null;
    LastError?: string | null;
}

export interface PiCompanionInfo {
    Configured: boolean;
    Host?: string | null;
    Reachable: boolean;
    LatencyMs?: number | null;
    LastError?: string | null;
}

File: BetterPlacemaking.SERVER/Models/JetsonDTOs/HealthReport.cs
using System.Collections.Generic;
using Google.Cloud.Firestore;

namespace BetterPlacemaking.Models.JetsonDTOs
{
    [FirestoreData]
    public class HealthReport
    {
        [FirestoreProperty]
        public long Timestamp { get; set; }

        [FirestoreProperty]
        public Dictionary<string, ServiceStatus>? Services { get; set; }

        [FirestoreProperty]
        public Dictionary<string, CameraInfo>? Cameras { get; set; }

        [FirestoreProperty]
        public SystemInfo? System { get; set; }

        [FirestoreProperty]
        public Dictionary<string, IntrinsicsCalibrationState>? IntrinsicsCalibration { get; set; }

        [FirestoreProperty]
        public Dictionary<string, LidarHealthInfo>? Lidars { get; set; }

        [FirestoreProperty]
        public PiCompanionHealthInfo? PiCompanion { get; set; }
    }

    [FirestoreData]
    public class LidarHealthInfo
    {
        [FirestoreProperty]
        public string? SensorId { get; set; }

        [FirestoreProperty]
        public bool Connected { get; set; }

        [FirestoreProperty]
        public string? DevicePath { get; set; }

        [FirestoreProperty]
        public string? UsbHint { get; set; }

        [FirestoreProperty]
        public string? LastError { get; set; }
    }

    [FirestoreData]
    public class PiCompanionHealthInfo
    {
        [FirestoreProperty]
        public bool Configured { get; set; }

        [FirestoreProperty]
        public string? Host { get; set; }

        [FirestoreProperty]
        public bool Reachable { get; set; }

        [FirestoreProperty]
        public int LatencyMs { get; set; }

        [FirestoreProperty]
        public string? LastError { get; set; }
    }

    [FirestoreData]
    public class IntrinsicsCalibrationState
    {
        [FirestoreProperty]
        public string? Status { get; set; }  // "collecting" | "computing" | "done" | "failed"

        [FirestoreProperty]
        public int SightingsCollected { get; set; }

        [FirestoreProperty]
        public List<int>? CoverageGrid { get; set; }  // flat list of 0/1 per cell

        [FirestoreProperty]
        public string? SuggestedRegion { get; set; }  // e.g. "top-left", null when all covered

        [FirestoreProperty]
        public string? SuggestedTilt { get; set; }

        [FirestoreProperty]
        public double CurrentRmse { get; set; }
    }

    [FirestoreData]
    public class CameraInfo
    {
        [FirestoreProperty]
        public string? Mac { get; set; }

        [FirestoreProperty]
        public string? Ip { get; set; }

        [FirestoreProperty]
        public List<int>? Resolution { get; set; }

        [FirestoreProperty]
        public bool Enabled { get; set; }
    }

    [FirestoreData]
    public class ServiceStatus
    {
        [FirestoreProperty]
        public string? Active { get; set; }

        [FirestoreProperty]
        public string? Sub { get; set; }
    }

    [FirestoreData]
    public class SystemInfo
    {
        [FirestoreProperty]
        public GpuInfo? Gpu { get; set; }

        [FirestoreProperty]
        public MemoryInfo? Memory { get; set; }

        [FirestoreProperty]
        public List<DiskPartitionInfo>? Disk { get; set; }

        [FirestoreProperty]
        public double CpuTemperatureC { get; set; }
    }

    [FirestoreData]
    public class DiskPartitionInfo
    {
        [FirestoreProperty]
        public string? Path { get; set; }

        [FirestoreProperty]
        public int TotalMb { get; set; }

        [FirestoreProperty]
        public int UsedMb { get; set; }

        [FirestoreProperty]
        public int FreeMb { get; set; }

        [FirestoreProperty]
        public int UsePct { get; set; }

        [FirestoreProperty]
        public string? Status { get; set; }  // "ok" | "warning" | "critical"

        [FirestoreProperty]
        public int DeletedFiles { get; set; }
    }

    [FirestoreData]
    public class GpuInfo
    {
        [FirestoreProperty]
        public int UtilizationPct { get; set; }

        [FirestoreProperty]
        public int FrequencyMhz { get; set; }

        [FirestoreProperty]
        public double TemperatureC { get; set; }
    }

    [FirestoreData]
    public class MemoryInfo
    {
        [FirestoreProperty]
        public int UsedMb { get; set; }

        [FirestoreProperty]
        public int TotalMb { get; set; }
    }
}
